MantidPlot/src/ProjectRecoveryGUIs/RecoveryThread.cpp

```diff
@@ -0,0 +1,22 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "RecoveryThread.h"
+#include "ProjectRecovery.h"
+
+bool RecoveryThread::getFailedRun() { return m_failedRunInThread; }
+
+void RecoveryThread::setCheckpoint(Poco::Path checkpoint) {
```

Comment:
Is it worth passing `checkpoint` as `const Poco::Path &`?

Reply:
Should't be a problem with concurrency as it keeps a copy in the thread, so yes.